python/pyarrow/ipc.pxi

```diff
@@ -399,21 +399,23 @@ cdef class MessageReader(_Weakrefable):
                         "instead.".format(self.__class__.__name__))
 
     @staticmethod
-    def open_stream(source):
+    def open_stream(source, use_memory_map=False):
```

Comment:
I think `user_memory_map` here is misleading for a user.
`source` can be a `PythonFile` or `NativeFile`, in such case the memory map option won't do anything. The option is only effective if the `source` is a string.

Given that the user can pass a `MemoryMappedFile` if they want to explicitly use `memory_map`. I think we should remove the option and explicitly document to use `MemoryMappedFile` as the source if you want to use memory mapping.

Reply:
Same applies for the other cases where `_get_input_stream` is used and a `use_memory_map` option was added.